go/database/mpt/proof_test.go

```diff
@@ -274,6 +274,48 @@ func TestCreateWitnessProof_CannotCreateProof_FailingNodeSources(t *testing.T) {
 	}
 }
 
+func TestCreateWitnessProof_SourceError_All_Paths(t *testing.T) {
+	injectedErr := errors.New("injected error")
+	address := common.Address{1}
+
+	for _, config := range allMptConfigs {
+		t.Run(config.Name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			ctxt := newNiceNodeContextWithConfig(t, ctrl, config)
+
+			addressNibbles := AddressToNibblePath(address, ctxt)
+			desc := &Branch{
+				children: Children{
+					addressNibbles[0]: &Branch{
+						dirtyChildHashes: []int{1},
```

Comment:
should this be the following?
```suggestion
						dirtyChildHashes: []int{addressNibbles[1]},
```

Reply:
no, this marks a child dirty, this is not the nibble path